src/org/zaproxy/zap/extension/httpsinfo/HttpsInfoDialog.java

```diff
@@ -0,0 +1,342 @@
+		this.centerFrame();
+		init();
+		addComponentsToPane(getContentPane());
+		this.pack();
+		this.setVisible(true);
+		try {
+			initSession(getTarget());
+		} catch (DVException e) {
+			String warnMsg = "An exception has prevented connection to: " + getTarget().toString() + ". Caused by: "
+					+ e.getCause(); 
+			LOGGER.warn(warnMsg);
+			general.setText(warnMsg);
+			return;
+		}
+		setDvEng(getSession());
+		updateContent();
+	}
+	
+	private JTextArea getGeneralTextArea() {
+		return general;
+	}
+	
+	private JTextArea getCipherSuitesTextArea() {
+		return cipherSuites;
+	}
+
+	public void init() {
+		checkProxyChainEnabled();
+		
+		initGeneral();
+		initCipherSuites();
+	}
+	
+	private void initGeneral() {
+		general = new JTextArea();
+		general.setEditable(false);
+		general.setRows(12);
+		general.setColumns(70);
+		general.setLineWrap(true);
+	}
+	
+	private void initCipherSuites() {
+		cipherSuites = new JTextArea();
+		cipherSuites.setEditable(false);
+		cipherSuites.setRows(15);
+		cipherSuites.setColumns(70);
+	}
+	
+	private void setTarget(HttpMessage msg) {
+		try {
+			this.target = new URL(msg.getRequestHeader().getURI().toString());
+		} catch (MalformedURLException e) {
+			LOGGER.warn("An exception occurred while attempting to set the target", e);
+		}
+	}
+	
+	private URL getTarget() {
+		if (target == null) {
+			LOGGER.warn("Somehow the target was not set, when we tried to use it.");
+			return null;
+		}
+		return target;
+	}
+	
+	private void setBaseMessage(HttpMessage msg) {
+		this.baseMessage = msg;
+	}
+	
+	private HttpMessage getBaseMessage() {
+		return baseMessage;
+	}
+	
+	private void initSession(URL target) throws DVException {
+		try {
+			this.session = DVFactory.initializeSession(target);
+		} catch (DVException e) {
+			throw new DVException("An exception occurred while initializing the DV session. " + e.getMessage(),
+					e.getCause());
+		}
+	}
+	
+	private IDVSession getSession() {
+		return session;
+	}
+	
+	private void setDvEng(IDVSession session) {
+		this.dvEng = DVFactory.getDVEng(session);
+	}
+	
+	private IDVEng getDvEng() {
+		return dvEng;
+	}
+
+	public void updateContent() {
+		try {
+			showGeneral();
+		} catch (DVException e) {
+			String generalException = "An exception occurred while updating the General panel. ";
+			LOGGER.warn(generalException, e);
+			getGeneralTextArea().setText(generalException + e.getMessage());
+		}
+		try {
+			showCipherSuites();
+		} catch (DVException e) {
+			String cipherSuitesException = "An exception occurred while updating the CipherSuites Panel. ";
+			LOGGER.warn(cipherSuitesException, e);
+			getCipherSuitesTextArea().setText(cipherSuitesException + e.getMessage());
+		}
+	}
+
+	public void showGeneral() throws DVException {
+		StringBuilder content = new StringBuilder(
+				Constant.messages.getString("httpsinfo.general.server.leadin", target.getHost()));
+		if (getDvEng().getCertificate() == null) {
```

Comment:
- This needs to be outside the EDT, if it takes too much time it will freeze the UI (i.e. no repaints will be done and we can't interact with the GUI).
 - Does it worth trying again? (I'd expect that to work at the first try.)